lib/db/mem.js

```diff
@@ -231,16 +233,26 @@ module.exports = function (log, error) {
   }
 
   function updateDeviceRecord (device, deviceInfo, deviceKey) {
-    var session
-    var sessionKey = (deviceInfo.sessionTokenId || '').toString('hex')
-
+    // Prevent multiple device records from linking to the same sessionToken.
+    let session
+    const sessionKey = (deviceInfo.sessionTokenId || '').toString('hex')
     if (sessionKey) {
```

Comment:
> Since we do not yet allow a deviceRecord to contain both a refreshTokenId and
> a sessionTokenId, should we enforce that?

IIUC we intend to allow this at some point in the not too distant future, so I don't think we should go out of our way to prevent it in the meantime.   I can't think of a way where it would become true by accident.

> Would a device with neither be considered an orphaned device?

Yes; going by the current style of the code I would just avoid reading those out of the db, essentially pretending they don't exist.

Reply:
(In fact I believe the current logic commented with "zombie devices" will avoid reading them out of the db).